test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Binders/MetadataBindingUtilsTests.cs

```diff
@@ -50,6 +51,86 @@ public void IfTypesCannotPromoteErrorIsThrown()
             Action convertMethod = () => MetadataBindingUtils.ConvertToTypeIfNeeded(node, targetType);
             convertMethod.Throws<ODataException>(ODataErrorStrings.MetadataBinder_CannotConvertToType(node.TypeReference.FullName(), targetType.FullName()));
         }
+
+        [Fact]
+        public void IfTypePromotionNeeded_SourceIsIntegralMemberValueAndTargetIsEnum_ConstantNodeIsCreated()
+        {
+            // Arrange
+            var enumValue = 3;
+
+            SingleValueNode source = new ConstantNode(enumValue);
+            IEdmTypeReference targetTypeReference = new EdmEnumTypeReference(WeekDayEmumType, false);
+
+            // Act
+            var result = MetadataBindingUtils.ConvertToTypeIfNeeded(source, targetTypeReference);
+
+            // Assert
+            var member = WeekDayEmumType.Members.First(m => m.Value.Value == enumValue);
+            result.ShouldBeEnumNode(WeekDayEmumType, member.Name);

```

Comment:
I think we should also assert the actual expected member name, just to make sure we're getting the enum value that we expected.

Reply:
@habbes 

Yes, this has been done. I have added this `assertion` to all the tests that I have added for this case

[MetadataBindingUtilsTests.cs](https://github.com/OData/odata.net/blob/fix/2542-filter-with-enum-integer/test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Binders/MetadataBindingUtilsTests.cs)